core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt

```diff
@@ -37,9 +41,29 @@ fun detectConflicts(trainRequirements: List<TrainRequirements>): List<Conflict>
 interface IncrementalConflictDetector {
     fun checkConflicts(): List<Conflict>
 
+    fun checkConflicts(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>
+    ): List<Conflict>
+
     fun checkSpacingRequirement(req: SpacingRequirement): List<Conflict>
 
     fun checkRoutingRequirement(req: RoutingRequirement): List<Conflict>
+
+    fun minDelayWithoutConflicts(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>
+    ): Double
+
+    fun maxDelayWithoutConflicts(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>
+    ): Double
+
+    fun timeOfNextConflict(
+        spacingRequirements: List<SpacingRequirement>,
+        routingRequirements: List<RoutingRequirement>
+    ): Double
```

Comment:
wouldn't it be faster to return all properties at once? I can think of having a analyzeConflicts() method which would return an object with all of these properties